src/components/Dialog/index.web.tsx

```diff
@@ -37,22 +40,32 @@ export function Outer({
     setDialogIsOpen(control.id, true)
   }, [setIsOpen, setDialogIsOpen, control.id])
 
-  const close = React.useCallback(async () => {
+  const onCloseInner = React.useCallback(async () => {
     setIsVisible(false)
     await new Promise(resolve => setTimeout(resolve, 150))
     setIsOpen(false)
     setIsVisible(true)
     setDialogIsOpen(control.id, false)
     onClose?.()
-  }, [onClose, setIsOpen, setDialogIsOpen, control.id])
+  }, [control.id, onClose, setDialogIsOpen])
+
+  const close = React.useCallback<DialogControlProps['close']>(
+    cb => {
+      if (cb && typeof cb === 'function') {
+        cb()
+      }
+      onCloseInner()
+    },
```

Comment:
Fine to make this a separate method imo, but let's also wrap in a try/catch to match native impl?

Reply:
Oh I see, types, makes sense to have seprate 👍